src/applications/apps.ts

```diff
@@ -131,7 +148,7 @@ export function registerApplication(
     );
 
   apps.push(
-    assign(
+    Object.assign(
```

Comment:
This is a runtime change that should be mostly equivalent and perhaps a bit faster (browser implementation vs userland implementation). IE10/11 didn't support Object.assign. Since [single-spa 7 won't support IE](https://single-spa.js.org/blog/2023/11/27/single-spa-6#browser-support--ie11), I switched it over. The typing benefit is that Object.assign knows the resultant object better than my handcrafted js-implementation of `assign()`.